next.config.js

```diff
@@ -26,6 +30,49 @@ module.exports = withCSS(withSourceMaps({
     APP_VERSION: packageJson.version,
     UNLY_SIMPLE_LOGGER_ENV: process.env.APP_STAGE, // Used by @unly/utils-simple-logger - Fix missing staging logs because it believes we're in production
   },
+  experimental: {
+    redirects() {
+      const redirects = [
+        {
+          // Redirect root link with trailing slash to non-trailing slash, avoids 404 - See https://github.com/zeit/next.js/discussions/10651#discussioncomment-8270
+          source: '/:locale/',
+          destination: '/:locale',
+          permanent: process.env.APP_STAGE !== 'development', // Do not use permanent redirect locally to avoid browser caching when working on it
+        },
+        {
+          // Redirect link with trailing slash to non-trailing slash (any depth), avoids 404 - See https://github.com/zeit/next.js/discussions/10651#discussioncomment-8270
+          source: '/:locale/:path*/',
+          destination: '/:locale/:path*',
+          permanent: process.env.APP_STAGE !== 'development', // Do not use permanent redirect locally to avoid browser caching when working on it
+        },
+      ];
+
+      if (process.env.APP_STAGE === 'development') {
+        console.info('Using experimental redirects:', redirects);
+      }
+
+      return redirects;
+    },
+    rewrites() {
+      const rewrites = [
+        {
+          // XXX Doesn't work locally (maybe because of rewrites), but works online
+          source: '/',
+          destination: '/api/autoRedirectToLocalisedPage',
+        },
+        {
+          source: `/:locale((?!${supportedLocales.join('|')})[^/]+)(.*)`,
```

Comment:
I tried to use your technique inside my project but it doesn't work for static assets inside `/public` folder. For example a requested static image is in 404 because it is redirected with locale path. Am I missing something in the configuration ?